src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java

```diff
@@ -657,23 +657,30 @@ public void deleteLocalRepository(URL repoUrl, String targetPath) {
      * @return path to zip file.
      */
     public Path zipRepository(Repository repo) throws IOException {
-        return zipRepository(repo, REPO_CLONE_PATH);
+        return zipRepository(repo, REPO_CLONE_PATH, false);
     }
 
     /**
      * Zip the content of a git repository.
      *
      * @param repo Local Repository Object.
      * @param targetPath path where the repo is located on disk
+     * @param hideStudentName option to hide the student name for the zip file
      * @throws IOException if the zipping process failed.
      * @return path to zip file.
      */
-    public Path zipRepository(Repository repo, String targetPath) throws IOException {
+    public Path zipRepository(Repository repo, String targetPath, boolean hideStudentName) throws IOException {
         String[] repositoryUrlComponents = repo.getParticipation().getRepositoryUrl().split(File.separator);
         ProgrammingExercise exercise = repo.getParticipation().getProgrammingExercise();
         String courseShortName = exercise.getCourseViaExerciseGroupOrCourseMember().getShortName().replaceAll("\\s", "");
-        // take the last component
-        String zipRepoName = courseShortName + "-" + repositoryUrlComponents[repositoryUrlComponents.length - 1] + ".zip";
+        String zipRepoName;
+        if (hideStudentName) {
+            zipRepoName = courseShortName + "-" + repositoryUrlComponents[repositoryUrlComponents.length - 2].toLowerCase() + "-student-submission" + ".zip";
+        }
+        else {
+            // take the last component
+            zipRepoName = courseShortName + "-" + repositoryUrlComponents[repositoryUrlComponents.length - 1] + ".zip";
+        }
```

Comment:
Can you assign `repositoryUrlComponents[n]` to a variable with a meaningful name?
Would help if we need to touch this again in a week or so :D

Reply:
I added detailed comments what we are using in `d4a225b`